### ISSUE TITLE
Blob is missing arrayBuffer() and other properties

### ISSUE THREAD

**Issue by Dealerpriest** [microsoft/TypeScript#35229]
[Blob interface](https://github.com/microsoft/TypeScript/blob/75301c8e2ce498359a6b33c3f9c9a6a1bd5980c0/lib/lib.dom.d.ts#L2519) is missing several properties documented in the spec over at at [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Blob). In my case I need the arrayBuffer() function.

**TypeScript Version:**  3.8.0-dev.20191119

**Search Terms:**
Blob, arrayBuffer(), Blob.arrayBuffer()

**Code**
```ts
myBlob: Blob = new Blob();
myBlob.arrayBuffer();
```

**Expected behavior:**
typescript should not give an error when using the arrayBuffer() function on a Blob instance.

**Actual behavior:**
typescript gives error.

**Playground Link:** 
[Playground Link](http://www.typescriptlang.org/play/?ts=3.8.0-dev.20191119#)

**Comment by MartinJohns** [comment]
The support according to the [browser compatibility](https://developer.mozilla.org/en-US/docs/Web/API/Blob/arrayBuffer#Browser_compatibility) is fairly bad. Also `arrayBuffer()` is only part of a [working draft specification](https://developer.mozilla.org/en-US/docs/Web/API/Blob/arrayBuffer#Specifications).

TypeScript usually only adds support when there's enough browser support and if the specification is done.

**Comment by MartinJohns** [comment]
Duplicate of #34652.

**Comment by Dealerpriest** [comment]
I see. Sorry for posting duplicate. I searched the repository's issues but for some reason didn't find that one 😩 
I managed to make use of declaration merging by putting the following into `declarations.d.ts`:
```
interface Blob {
  arrayBuffer(): Promise<ArrayBuffer>;
}
```

Cheers!

**Comment by ascdi** [comment]
For me it just wasn't there and adding it in declarations didn't work.
I had to use this backwards compatible method ([found here](https://stackoverflow.com/a/66167881/532495)) to solve the issue:

```
async function blobToArrayBuffer(blob: Blob): Promise<ArrayBuffer> {
    if ('arrayBuffer' in blob) return await blob.arrayBuffer();
    
    return new Promise((resolve, reject) => {
        const reader = new FileReader();
        reader.readAsArrayBuffer(blob);
        reader.onload = () => resolve(reader.result as ArrayBuffer);
        reader.onerror = () => reject();
    });
}
```

**Comment by ascdi** [comment]
The `declarations.d.ts` solution didn't work for me, the method simply wasn't there.

I now use this backwards compatible helper function ([found here](https://stackoverflow.com/a/66167881/532495)):

```
export async function blobToArrayBuffer(blob: Blob): Promise<ArrayBuffer> {
  if ('arrayBuffer' in blob) return await blob.arrayBuffer();

  return new Promise((resolve, reject) => {
    const reader = new FileReader();
    reader.readAsArrayBuffer(blob);
    reader.onload = () => resolve(reader.result as ArrayBuffer);
    reader.onerror = (e) => reject(e);
  });
}
```
